### PR TITLE
Fix: Better sanity check of hostname when adding device.

### PR DESCRIPTION
**Pull request by Zmegolaz**
Fixes a bug where you can add devices without hostname.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

### CODE REVIEW

**Comment by murrant** [comment]
You forgot to check for IPs.

`IP::isValid($hostname)` might be handy ;)

**Comment by Zmegolaz** [comment]
is_valid_hostname() works for IPs too, since it allows dots and `[a-z\d]` mixed. Or is there a risk it will be modified to do a more strict check later on?

**Comment by laf** [comment]
Should we not be assigning $hostname = clean($_POST['hostname']); before all the checks then checking $hostname otherwise we may have a case where is_valid_hostname() is true before clean() is applied?

**Comment by murrant** [comment]
@Zmegolaz Does it work for IPv6 addresses?

**Comment by Zmegolaz** [comment]
Very good points, both of you. I've updated it.

File: html/pages/addhost.inc.php [hunk 1]
```diff
@@ -10,7 +11,11 @@
     exit;
 }
 
-$hostname = isset($_POST['hostname']) ? clean($_POST['hostname']) : false;
+$hostname = (isset($_POST['hostname']) && clean($_POST['hostname']) != "") ? clean($_POST['hostname']) : false;
```
Comment (murrant) [inline html/pages/addhost.inc.php]: Can you change this from a ternary to a regular if else?  That would make it more readable.

File: html/pages/addhost.inc.php [hunk 2]
```diff
@@ -10,7 +11,11 @@
...
+if ($hostname !== false && !is_valid_hostname($hostname) && !IP::isValid($hostname)) {
+    $hostname = false;
+    print_error('Invalid hostname.');
```
Comment (murrant) [inline html/pages/addhost.inc.php]: The text should probably be `Invalid hostname or IP.`

**Pull request merged**